workspace_tools/host_tests/wfi_auto.py

```diff
@@ -0,0 +1,37 @@
+"""
+mbed SDK
+Copyright (c) 2011-2013 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import sys
+import uuid
+import time
+from sys import stdout
+
+class WFITest():
+
+    def test(self, selftest):
+        result = True
+        c = selftest.mbed.serial_readline()
```

Comment:
Tests on Nucleos may fail because you are assuming board will not 'eat' first few characters it should print.
Unfortunately it could happen so I would just in your test read values and make sure they are incrementing integers.